graphql-protocol/src/main/java/com/linecorp/armeria/server/graphql/protocol/AbstractGraphqlService.java

```diff
@@ -90,6 +98,45 @@ protected HttpResponse doPost(ServiceRequestContext ctx, HttpRequest request) th
             return unsupportedMediaType();
         }
 
+        // See https://github.com/jaydenseric/graphql-multipart-request-spec/blob/master/readme.md
+        if (contentType.is(MediaType.MULTIPART_FORM_DATA)) {
+            return HttpResponse.from(FileAggregatedMultipart.aggregateMultipart(ctx, request)
+                                                            .thenApply(multipart -> {
+                try {
+                    final ListMultimap<String, String> multipartParams = multipart.params();
+                    final String operationsParam = getValueFromMultipartParam("operations", multipartParams);
+                    final String mapParam = getValueFromMultipartParam("map", multipartParams);
+
+                    final Map<String, Object> operations = parseJsonString(operationsParam, JSON_MAP);
+                    final Map<String, List<String>> map = parseJsonString(mapParam, MAP_PARAM);
+                    final String query = toStringFromJson("query", operations.get("query"));
+                    if (Strings.isNullOrEmpty(query)) {
+                        return HttpResponse.of(HttpStatus.BAD_REQUEST, MediaType.PLAIN_TEXT,
+                                               "Missing query");
```

Comment:
If I change this, I think this same message should also be changed, but it could be ambiguous because the additionally referenced values are different. Do you mean to ignore this and change this message?

Reply:
> because the additionally referenced values are different.

Oops, I think I don't get this. 😅 
Could you elaborate more on this? 🙏

Reply:
It meant the same message was in multiple places 🤣 

- https://github.com/line/armeria/blob/master/graphql-protocol/src/main/java/com/linecorp/armeria/server/graphql/protocol/AbstractGraphqlService.java#L63
- https://github.com/line/armeria/blob/master/graphql-protocol/src/main/java/com/linecorp/armeria/server/graphql/protocol/AbstractGraphqlService.java#L108
- https://github.com/line/armeria/blob/master/graphql-protocol/src/main/java/com/linecorp/armeria/server/graphql/protocol/AbstractGraphqlService.java#L138

Reply:
Ah, thanks for the links. 😄 
I personally prefer to change all of them because that will help the clients easily recognize what is wrong with the request.